rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/core/RMCore.java

```diff
@@ -1669,6 +1665,22 @@ public BooleanWrapper shutdown(boolean preempt) {
                 removeAllNodes(entry.getKey(), preempt, true);
                 entry.getValue().shutdown(caller);
             }
+
+            boolean atLeastOneAlive = false;
```

Comment:
you should extract the wait logic in a dedicated method

Reply:
Done. This will be `waitForAllNodeSourcesToBeShutdown`.